cmd/decode.go

```diff
@@ -95,21 +104,71 @@ func validateAndRun() error {
 	return run(options.metaOnly, outMediaType, in, os.Stdout)
 }
 
+// Original requirement from etcd WAL log analysis tool etcd-dump-logs,
+// to serve as a decoder to decode k8s objects from each entry of the etcd WAL log.
+// Read data from os.Stdin, decode object line by line, and print out in os.Stdout
+func runInBatchMode(metaOnly bool, outMediaType string, out io.Writer) error {
+	var err error
+	batchmode := true
+	inputReader := bufio.NewReader(os.Stdin)
+	for {
+		input, err := inputReader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			fmt.Printf("EOF\n")
+			return nil
+		}
+		if err != nil {
+			fmt.Printf("Read Error: %s\n", err)
+			return err
+		}
+
+		input = stripNewline(input)
+		decodedinput, err := hex.DecodeString(string(input))
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		inMediaType, decodedinput, err := encoding.DetectAndExtract(decodedinput)
+		if err != nil {
+			fmt.Printf("AUGER_ERROR:%s|\n", err)
+			continue
+		}
+
+		if metaOnly {
+			err = encoding.DecodeSummary(inMediaType, batchmode, decodedinput, out)
+			if err != nil {
+				fmt.Printf("|AUGER_ERROR:%s\n", err)
+				continue
+			}
+			fmt.Printf("|AUGER_ERROR:%s\n", err)
+			continue
+		}
+
+		err = encoding.Convert(inMediaType, outMediaType, batchmode, decodedinput, out)
+		if err != nil {
+			fmt.Printf("|AUGER_ERROR:%s\n", err)
+			continue
+		}
+	}
+	return err
```

Comment:
`return nil` ?  This will return the last error encountered in the `encoding.DecodeSummary` and `encoding.Convert` as written, but we already report those errors in the output. Do we want auger to print a final error and return exit status 1 when any of the decodes fail? I can imagine arguments both ways, but if we do, we should return a more descriptive error.